File: service/src/main/java/com/ooftf/service/java/SafeOnClickListener.java
package com.ooftf.service.java;

import android.view.View;

public abstract class SafeOnClickListener implements View.OnClickListener {

    long last = 0L;

    @Override
    final public void onClick(View view) {
        long current = System.currentTimeMillis();
        if (current - last > safeMillis()) {
            safeOnClick(view);
            last = current;
        }
    }

    public abstract void safeOnClick(View view);

    public long safeMillis() {
        return 1000;
    }
}

File: service/src/main/java/com/ooftf/service/engine/SafeOnClickListener.kt
package com.ooftf.service.engine

import android.view.View

/**
 * Created by 99474 on 2017/11/17 0017.
 */
abstract class SafeOnClickListener : View.OnClickListener {
    var last = 0L
    override fun onClick(view: View) {
        var current = System.currentTimeMillis()
        if (current - last > safeMillis()) {
            safeOnClick(view)
            last = current
        }
    }

    abstract fun safeOnClick(view: View)
    open fun safeMillis(): Long = 1000
}